### PR TITLE
CI: add Alpine/musl

### PR DESCRIPTION
**Pull request by thesamesam**

### CODE REVIEW

File: .github/workflows/compiler-zoo.yml [hunk 1]
```diff
@@ -12,7 +12,45 @@ on: [push]
 permissions:
   contents: read
 
+# This has to be a separate job, it seems, because we want to use a
```
Comment (paulidale) [inline .github/workflows/compiler-zoo.yml:15]: A thought: would this be better placed in the os-zoo.yml file?

It seems to be more about Alpine than a compiler.  If not now, then in the future when others have caught up.

Comment (thesamesam) [reply .github/workflows/compiler-zoo.yml:15]: Ah, good idea!

**Comment by t8m** [comment]
Please drop the first commit - that one is handled by https://github.com/openssl/openssl/pull/19581

**Comment by thesamesam** [comment]
> Please drop the first commit - that one is handled by #19581

Done (although it didn't exist at the time 😉)

**Review by t8m — APPROVED** [review]
Approving. Assuming that you've verified the CI job passes.

**Comment by thesamesam** [comment]
> Approving. Assuming that you've verified the CI job passes.

Yeah, you can see it at https://github.com/thesamesam/openssl/actions/runs/3374981533. Thanks a bunch!

**Review by paulidale — APPROVED** [review]

**Comment by thesamesam** [comment]
> Looks good. Adding enable-quic might be a nice idea since that touches network code and testing against musl would be good.
> 

I'll do that in a follow up if that's alright. It's a good idea.

> Is there a reason we have OPENSSL_USE_IPV6=0 here?

I think I hit a variant of https://github.com/openssl/openssl/pull/19181 and I meant to go back to it. I'll look at it in the follow up for QUIC?

**Comment by hlandau** [comment]
@thesamesam Sure, sounds good.

**Comment by thesamesam** [comment]
Thanks!

**Comment by t8m** [comment]
Merged to master branch. The patch does not apply to 3.1 however I'll create PR to sync the 3.1 CI with master later.

Thank you for your contribution.

**Comment by thesamesam** [comment]
Thank you!

**Pull request closed without merging**